redfish/computersystem.go

```diff
@@ -635,6 +671,55 @@ func (computersystem *ComputerSystem) SecureBoot() (*SecureBoot, error) {
 	return GetSecureBoot(computersystem.Client, computersystem.secureBoot)
 }
 
+// SetBoot set a boot object based on a payload request
+func (computersystem *ComputerSystem) SetBoot(b Boot) (bool, error) {
+	type temp struct {
+		Boot Boot
+	}
+	t := temp{
+		Boot: b,
+	}
+
+	payload, err := json.Marshal(t)
+	if err != nil {
+		return false, err
+	}
+
+	resp, err := computersystem.Client.Patch(computersystem.ODataID, payload)
+	if err != nil {
+		return false, err
+	}
+	defer resp.Body.Close()
+
+	return true, nil
+}
+
+// Reset This action shall perform a reset of the ComputerSystem.  For systems which implement ACPI Power Button
+// functionality, the PushPowerButton value shall perform or emulate an ACPI Power Button push.  The ForceOff
+// value shall remove power from the system or perform an ACPI Power Button Override (commonly known as a
+// 4-second hold of the Power Button).  The ForceRestart value shall perform a ForceOff action followed by a On action.
+func (computersystem *ComputerSystem) Reset(r ResetType) (bool, error) {
+	type temp struct {
+		ResetType ResetType
+	}
+	t := temp{
+		ResetType: r,
+	}
+
+	payload, err := json.Marshal(t)
+	if err != nil {
+		return false, err
+	}
+
+	resp, err := computersystem.Client.Post(computersystem.Actions.ComputerSystemReset.Target, payload)
+	if err != nil {
+		return false, err
+	}
+	defer resp.Body.Close()
+
+	return true, nil
```

Comment:
Is there a reason for wanting a bool returned as well rather than just the normal check of "if err != nil"?